x/liquidity/keeper/liquidity_pool.go

```diff
@@ -73,16 +74,30 @@ func (k Keeper) CreatePool(ctx sdk.Context, msg *types.MsgCreatePool) (types.Poo
 	poolCreator := msg.GetPoolCreator()
 	poolCreatorBalances := k.bankKeeper.GetAllBalances(ctx, poolCreator)
 
-	if !poolCreatorBalances.IsAllGTE(msg.DepositCoins) {
-		return types.Pool{}, types.ErrInsufficientBalance
+	for _, coin := range msg.DepositCoins {
+		if coin.Amount.LT(params.MinInitDepositAmount) {
+			return types.Pool{}, sdkerrors.Wrapf(
+				types.ErrLessThanMinInitDeposit, "deposit coin %s is smaller than %s", coin, params.MinInitDepositAmount)
+		}
 	}
 
 	for _, coin := range msg.DepositCoins {
-		if coin.Amount.LT(params.MinInitDepositAmount) {
-			return types.Pool{}, types.ErrLessThanMinInitDeposit
+		balance := k.bankKeeper.GetBalance(ctx, poolCreator, coin.Denom)
+		if balance.IsLT(coin) {
+			return types.Pool{}, sdkerrors.Wrapf(
+				types.ErrInsufficientBalance, "%s is smaller than %s", balance, coin)
```

Comment:
I wonder if there is going to be any issue if we combine the two for loops?

```go
for _, coin := range msg.DepositCoins {
    if coin.Amount.LT(params.MinInitDepositAmount) {
        return types.Pool{}, sdkerrors.Wrapf(
            types.ErrLessThanMinInitDeposit, "deposit coin %s is smaller than %s", coin, params.MinInitDepositAmount)
    }
    
    balance := k.bankKeeper.GetBalance(ctx, poolCreator, coin.Denom)
    if balance.IsLT(coin) {
        return types.Pool{}, sdkerrors.Wrapf(
            types.ErrInsufficientBalance, "%s is smaller than %s", balance, coin)
    }
}
```

Reply:
There will be no issue, only the order of errors might be affected.
So the checking order would be changed from: `X -> Y -> X -> Y`
to: `X -> X -> Y -> Y`

Reply:
But to keep consistency between other checks, and the size of `msg.DepositCoins` is small enough(=2), we can keep this logic as is.